app/models/device.rb

```diff
@@ -41,10 +41,22 @@ def filtered_checkins(args)
   def sanitize_checkins(sanitized, args)
     if args[:type] == 'address'
       sanitized = sanitized.map(&:reverse_geocode!) unless args[:action] == 'index' && args[:multiple_devices]
+      # if only one checkin in relation geocoding would turn into array so needs to be converted back sometimes
+      sanitized = checkins.where(id: sanitized[0].id) if sanitized.class.to_s == 'Array'
```

Comment:
As comment explains, not sure why this was happening, but when you request just one checkin and for it to be geocoded, for some reason sanitized would turn into an array after .map(&:reverse_geocode!)  and you can't use .select on an array, only on an active record collection thing. 

If you have more than one checkin the .map(&:reverse_geocode!) does not turn it into an array.